frontend/src/features/comparisons/statsSlice.ts

```diff
@@ -0,0 +1,37 @@
+import { createAsyncThunk, createSlice } from '@reduxjs/toolkit';
+import { RootState } from 'src/app/store';
+
+import { Statistics, StatsService } from 'src/services/openapi';
+
+const initialState: Statistics = {
+  active_users: {
+    total: 0,
+    joined_last_month: 0,
+    joined_last_30_days: 0,
+  },
+  polls: [],
+};
+
+export const fetchStats = createAsyncThunk(
+  'stats/fetchStats',
+  async (): Promise<Statistics> => {
+    return await StatsService.statsRetrieve();
+  }
+);
+
+const statsSlice = createSlice({
+  name: 'stats',
+  initialState,
+  reducers: {},
+  extraReducers: (builder) => {
+    // Refresh all statistics.
+    builder.addCase(fetchStats.fulfilled, (state, action) => {
+      state.active_users = action.payload.active_users;
+      state.polls = action.payload.polls;
+    });
+  },
+});
+
+export const selectStats = (state: RootState) => state.stats;
+
+export default statsSlice.reducer;
```

Comment:
It's not clear when exactly the "stats" object will be initialized with the values from the server. The current implementation seems to assume that `<HomeVideos>`  is responsible for calling `fetchStats()`. But that does not seem to be a correct assumption. For example, a user may directly access the comparison page without loading the home page first.  In this case, the "stats" will not be fetched, and the `<CollectiveGoalWeeklyProgress>` will show 0 comparison, without any request to the server.

I must say that I am not convinced of the benefit of using a Redux Slice in this case. I think we expect the stats values to be refreshed on every session anyway. So creating a custom hook to memorize the stats based on the current poll during the current session could be simpler to avoid this kind of bugs.

Reply:
We can use React Query and its cache and stale time features to optimize our backend calls. Would like to know the team's advice and feasibility on this as well, and feel free to let me know if some changes are to be made.

Reply:
Thanks for the feedback, I'll refactor this tomorrow.

> The current implementation seems to assume that <HomeVideos> is responsible for calling fetchStats(). But that does not seem to be a correct assumption. For example, a user may directly access the comparison page without loading the home page first. 

That's true. Instead we can fetch the statistics in the `<App>` component by using a hook, the same way we fetch the user's preferences.

> I must say that I am not convinced of the benefit of using a Redux Slice in this case. I think we expect the stats values to be refreshed on every session anyway. So creating a custom hook to memorize the stats based on the current poll during the current session could be simpler to avoid this kind of bugs.

Yes it's not useful indeed to save the data in the local storage as we want them to be refreshed for every session. It's still easily possible to manage the statistics global state with Redux without the need of saving the data in the local storage. The package `redux-persist` we use offers the possibility to declare [which states should be persisted](https://github.com/rt2zz/redux-persist#blacklist--whitelist) between sessions.

```javascript
const persistConfig = {
  key: 'root',
  storage,
  // The following line does the trick.
  blacklist: ['stats']
};
```

We could persist in the local storage user specific settings, and keep the rest outside. What do you think about it @amatissart ? 

I still see some benefits of using Redux to manage those two kind of global states (session-only and persisted), such as:
- having a unified interface that avoids to "re-invent the wheel"
- it "forces" the contributors to follow a documented path when they work on global states

Reply:
I have no strong opinion about this. Your approach looks good to me, if you think this pattern could indeed be easier to replicate for other features in the future.

Reply:
Alright, let's go.

> if you think this pattern could indeed be easier to replicate for other features in the future.

To be honest I'm not strongly convinced too. What motivates me is the fact that we (I?) have a good opportunity to discover what kind of limitations we will face with Redux, by actually using it.

Reply:
Hi @GresilleSiffle , I was trying to find something more robust to move forward with this approach.
I found out that `React Query` can be a good (future?) substitute for this one. Here's a clip that I believe touches on the exact obstacle that we might foresee: https://youtube.com/clip/Ugkxlyz_ErUUjudZYtywHwmyDVBub7fg6gfI 

Adding the link here for future references : )